src/main/scala/com/amazon/deequ/analyzers/Histogram.scala

```diff
@@ -125,13 +126,115 @@ case class Histogram(
   val MaximumAllowedDetailBins = 1000
+  val count_function = "count"
+  val sum_function = "sum"
+
+
+  def createAggregateFunction(function: String, aggregateColumn: String): AggregateFunction = {
+    function match {
+      case Histogram.count_function => Count
+      case Histogram.sum_function => Sum(aggregateColumn)
+      case _ => throw new IllegalArgumentException("Wrong aggregate function name: " + function)
+    }
+  }
+
+  sealed trait AggregateFunction {
+    def query(column: String, data: DataFrame): DataFrame
+
+    def total(data: DataFrame): Long
+
+    def aggregateColumn(): Option[String]
+
+    def function(): String
+  }
+
+  case object Count extends AggregateFunction {
+    override def query(column: String, data: DataFrame): DataFrame = {
+      data
+        .select(col(column).cast(StringType))
+        .na.fill(Histogram.NullFieldReplacement)
+        .groupBy(column)
+        .count()
+        .withColumnRenamed("count", Analyzers.COUNT_COL)
+    }
+
+    override def aggregateColumn(): Option[String] = None
+
+    override def function(): String = count_function
+
+    override def total(data: DataFrame): Long = {
+      data.count()
+    }
+  }
+
+  case class Sum(aggColumn: String) extends AggregateFunction {
+    override def query(column: String, data: DataFrame): DataFrame = {
+      data
+        .select(col(column).cast(StringType), col(aggColumn).cast(LongType))
+        .na.fill(Histogram.NullFieldReplacement)
+        .groupBy(column)
+        .sum(aggColumn)
+        .withColumnRenamed("count", Analyzers.COUNT_COL)
+    }
+
+    override def total(data: DataFrame): Long = {
+      data.groupBy().sum(aggColumn).first().getLong(0)
+    }
+
+    override def aggregateColumn(): Option[String] = {
+      Some(aggColumn)
+    }
+
+    override def function(): String = sum_function
+  }
+
+//  case class Min(aggColumn: String) extends AggregateFunction {
+//    override def query(column: String, data: DataFrame): DataFrame = {
+//      data
+//        .select(col(column).cast(StringType), col(aggColumn).cast(LongType))
+//        .na.fill(Histogram.NullFieldReplacement)
+//        .groupBy(column)
+//        .min(aggColumn)
+//        .withColumnRenamed("count", Analyzers.COUNT_COL)
+//    }
+//
+//    override def aggregateColumn(): Option[String] = {
+//      Some(aggColumn)
+//    }
+//
+//    override def function(): String = min
+//  }
+//
+//  case class Max(aggColumn: String) extends AggregateFunction {
+//    override def query(column: String, data: DataFrame): DataFrame = {
+//      data
+//        .select(col(column).cast(StringType), col(aggColumn).cast(LongType))
+//        .na.fill(Histogram.NullFieldReplacement)
+//        .groupBy(column)
+//        .max(aggColumn)
+//        .withColumnRenamed("count", Analyzers.COUNT_COL)
+//    }
+//
+//    override def aggregateColumn(): Option[String] = {
+//      Some(aggColumn)
+//    }
+//
+//    override def function(): String = max
+//  }
```

Comment:
Either we remove the commented out code, or we keep it as code and add tests. Otherwise, it adds bloat to the source code.

Reply:
I guess. I just forgot to remove this code. Will do